### ISSUE TITLE
Using ramsey/uuid-doctrine for entity's PK fails persisting entity with $I->haveInRepository

### ISSUE THREAD

**Issue by erop** [Codeception/Codeception#5691]
#### What are you trying to achieve?
I'm trying to execute test case which runs persisting in database with $I->haveInRepository
#### What do you get instead?
I have an error (see below)
IMO the reason of the issue is when `Doctrine2::debugEntityCreation()` is called there is a check agains `is_object()` for `$pk` being instance of `Ramsey\Uuid\Uuid` that leads to subsequent extracting primary keys from it that is delegated to Doctrine' EntityManger.

Downgrading to Codeception 3.1.0 eliminates issue. 
> Provide console output if related. Use `-vvv` mode for more details.

```bash
1) UserCest: Test Creating new User instance
 Test  tests/functional/UserCest.php:tryToTest

  [Doctrine\Common\Persistence\Mapping\MappingException] The class 'Ramsey\Uuid\Uuid' was not found in the chain configured namespaces App\Entity


Scenario Steps:

 1. $I->haveInRepository("App\\Entity\\User") at tests/functional/UserCest.php:13

#1  /Users/erop/Developer/temp/uuid-test/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/MappingException.php:22
#2  /Users/erop/Developer/temp/uuid-test/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/Driver/MappingDriverChain.php:87
#3  /Users/erop/Developer/temp/uuid-test/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:151
#4  /Users/erop/Developer/temp/uuid-test/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:304
#5  /Users/erop/Developer/temp/uuid-test/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:78
#6  /Users/erop/Developer/temp/uuid-test/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:183
#7  /Users/erop/Developer/temp/uuid-test/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:283
#8  Codeception\Module\Doctrine2->haveInRepository
#9  /Users/erop/Developer/temp/uuid-test/tests/_support/_generated/FunctionalTesterActions.php:2792
#10 /Users/erop/Developer/temp/uuid-test/tests/functional/UserCest.php:13
```
> Provide test source code if related
Whole simplest project code built from scratch can be found here https://github.com/erop/ramsey-uuid-in-codeception 
### Details

* Codeception version: 3.1.1
* PHP Version: 7.3.9
* Operating System: Mac OS X 10.14.6
* Installation type: Composer
* List of installed packages (`composer show`)
api-platform/api-pack               v1.2.0     A pack for API Platform
api-platform/core                   v2.4.7     Build a fully-featured hypermedia or GraphQL API in minutes
behat/gherkin                       v4.6.0     Gherkin DSL parser for PHP 5.3
codeception/codeception             3.1.1      BDD-style testing framework
codeception/phpunit-wrapper         8.0.4      PHPUnit classes used by Codeception
codeception/stub                    3.0.0      Flexible Stub wrapper for PHPUnit's Mock Builder
doctrine/annotations                v1.7.0     Docblock Annotations Parser
doctrine/cache                      v1.8.0     Caching library offering an object-oriented API for many cache backends
doctrine/collections                v1.6.2     PHP Doctrine Collections library that adds additional functionality on top of PHP arrays.
doctrine/common                     v2.11.0    PHP Doctrine Common project is a library that provides additional functionality that other ...
doctrine/dbal                       v2.9.2     Powerful PHP database abstraction layer (DBAL) with many features for database schema intro...
doctrine/doctrine-bundle            1.11.2     Symfony DoctrineBundle
doctrine/doctrine-cache-bundle      1.3.5      Symfony Bundle for Doctrine Cache
doctrine/doctrine-migrations-bundle v2.0.0     Symfony DoctrineMigrationsBundle
doctrine/event-manager              v1.0.0     Doctrine Event Manager component
doctrine/inflector                  v1.3.0     Common String Manipulations with regard to casing and singular/plural rules.
doctrine/instantiator               1.2.0      A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                      1.1.0      PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
doctrine/migrations                 2.1.1      PHP Doctrine Migrations project offer additional functionality on top of the database abstr...
doctrine/orm                        v2.6.3     Object-Relational-Mapper for PHP
doctrine/persistence                1.1.1      The Doctrine Persistence project is a set of shared interfaces and functionality that the d...
doctrine/reflection                 v1.0.0     Doctrine Reflection component
facebook/webdriver                  1.7.1      A PHP client for Selenium WebDriver
fig/link-util                       1.0.0      Common utility implementations for HTTP links
guzzlehttp/guzzle                   6.3.3      Guzzle is a PHP HTTP client library
guzzlehttp/promises                 v1.3.1     Guzzle promises library
guzzlehttp/psr7                     1.6.1      PSR-7 message implementation that also provides common utility methods
hoa/consistency                     1.17.05.02 The Hoa\Consistency library.
hoa/console                         3.17.05.02 The Hoa\Console library.
hoa/event                           1.17.01.13 The Hoa\Event library.
hoa/exception                       1.17.01.16 The Hoa\Exception library.
hoa/file                            1.17.07.11 The Hoa\File library.
hoa/iterator                        2.17.01.10 The Hoa\Iterator library.
hoa/protocol                        1.17.01.14 The Hoa\Protocol library.
hoa/stream                          1.17.02.21 The Hoa\Stream library.
hoa/ustring                         4.17.01.16 The Hoa\Ustring library.
jdorn/sql-formatter                 v1.2.17    a PHP SQL highlighting library
myclabs/deep-copy                   1.9.3      Create deep copies (clones) of your objects
nelmio/cors-bundle                  1.5.6      Adds CORS (Cross-Origin Resource Sharing) headers support in your Symfony2 application
nikic/php-parser                    v4.2.4     A PHP parser written in PHP
ocramius/package-versions           1.5.1      Composer plugin that provides efficient querying for installed package versions (no runtime...
ocramius/proxy-manager              2.2.3      A library providing utilities to generate, instantiate and generally operate with Object Pr...
phar-io/manifest                    1.0.3      Component for reading phar.io manifest information from a PHP Archive (PHAR)
phar-io/version                     2.0.1      Library for handling version information and constraints
phpdocumentor/reflection-common     2.0.0      Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock   4.3.2      With this component, a library can provide support for annotations via DocBlocks or otherwi...
phpdocumentor/type-resolver         1.0.1      A PSR-5 based resolver of Class names, Types and Structural Element Names
phpspec/prophecy                    1.8.1      Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage           7.0.8      Library that provides collection, processing, and rendering functionality for PHP code cove...
phpunit/php-file-iterator           2.0.2      FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template           1.2.1      Simple template engine.
phpunit/php-timer                   2.1.2      Utility class for timing
phpunit/php-token-stream            3.1.1      Wrapper around PHP's tokenizer extension.
phpunit/phpunit                     8.3.5      The PHP Unit Testing framework.
psr/cache                           1.0.1      Common interface for caching libraries
psr/container                       1.0.0      Common Container Interface (PHP FIG PSR-11)
psr/http-message                    1.0.1      Common interface for HTTP messages
psr/link                            1.0.0      Common interfaces for HTTP links
psr/log                             1.1.0      Common interface for logging libraries
ralouphie/getallheaders             3.0.3      A polyfill for getallheaders.
ramsey/uuid                         3.8.0      Formerly rhumsaa/uuid. A PHP 5.4+ library for generating RFC 4122 version 1, 3, 4, and 5 un...
ramsey/uuid-doctrine                1.5.0      Allow the use of a ramsey/uuid UUID as Doctrine field type.
sebastian/code-unit-reverse-lookup  1.0.1      Looks up which function or method a line of code belongs to
sebastian/comparator                3.0.2      Provides the functionality to compare PHP values for equality
sebastian/diff                      3.0.2      Diff implementation
sebastian/environment               4.2.2      Provides functionality to handle HHVM/PHP environments
sebastian/exporter                  3.1.2      Provides the functionality to export PHP variables for visualization
sebastian/global-state              3.0.0      Snapshotting of global state
sebastian/object-enumerator         3.0.3      Traverses array structures and object graphs to enumerate all referenced objects
sebastian/object-reflector          1.1.1      Allows reflection of object attributes, including inherited and non-public ones
sebastian/recursion-context         3.0.0      Provides functionality to recursively process PHP variables
sebastian/resource-operations       2.0.1      Provides a list of PHP built-in functions that operate on resources
sebastian/type                      1.1.3      Collection of value objects that represent the types of the PHP type system
sebastian/version                   2.0.1      Library that helps with managing the version number of Git-hosted PHP projects
symfony/asset                       v4.3.4     Symfony Asset Component
symfony/browser-kit                 v4.3.4     Symfony BrowserKit Component
symfony/cache                       v4.3.4     Symfony Cache component with PSR-6, PSR-16, and tags
symfony/cache-contracts             v1.1.5     Generic abstractions related to caching
symfony/config                      v4.3.4     Symfony Config Component
symfony/console                     v4.3.4     Symfony Console Component
symfony/css-selector                v4.3.4     Symfony CssSelector Component
symfony/debug                       v4.3.4     Symfony Debug Component
symfony/dependency-injection        v4.3.4     Symfony DependencyInjection Component
symfony/doctrine-bridge             v4.3.4     Symfony Doctrine Bridge
symfony/dom-crawler                 v4.3.4     Symfony DomCrawler Component
symfony/dotenv                      v4.3.4     Registers environment variables from a .env file
symfony/event-dispatcher            v4.3.4     Symfony EventDispatcher Component
symfony/event-dispatcher-contracts  v1.1.5     Generic abstractions related to dispatching event
symfony/expression-language         v4.3.4     Symfony ExpressionLanguage Component
symfony/filesystem                  v4.3.4     Symfony Filesystem Component
symfony/finder                      v4.3.4     Symfony Finder Component
symfony/flex                        v1.4.5     Composer plugin for Symfony
symfony/framework-bundle            v4.3.4     Symfony FrameworkBundle
symfony/http-foundation             v4.3.4     Symfony HttpFoundation Component
symfony/http-kernel                 v4.3.4     Symfony HttpKernel Component
symfony/inflector                   v4.3.4     Symfony Inflector Component
symfony/maker-bundle                v1.13.0    Symfony Maker helps you create empty commands, controllers, form classes, tests and more so...
symfony/mime                        v4.3.4     A library to manipulate MIME messages
symfony/polyfill-intl-idn           v1.12.0    Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-mbstring           v1.12.0    Symfony polyfill for the Mbstring extension
symfony/polyfill-php72              v1.12.0    Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73              v1.12.0    Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/process                     v4.3.4     Symfony Process Component
symfony/property-access             v4.3.4     Symfony PropertyAccess Component
symfony/property-info               v4.3.4     Symfony Property Info Component
symfony/routing                     v4.3.4     Symfony Routing Component
symfony/security-bundle             v4.3.4     Symfony SecurityBundle
symfony/security-core               v4.3.4     Symfony Security Component - Core Library
symfony/security-csrf               v4.3.4     Symfony Security Component - CSRF Library
symfony/security-guard              v4.3.4     Symfony Security Component - Guard
symfony/security-http               v4.3.4     Symfony Security Component - HTTP Integration
symfony/serializer                  v4.3.4     Symfony Serializer Component
symfony/service-contracts           v1.1.6     Generic abstractions related to writing services
symfony/stopwatch                   v4.3.4     Symfony Stopwatch Component
symfony/translation-contracts       v1.1.6     Generic abstractions related to translation
symfony/twig-bridge                 v4.3.4     Symfony Twig Bridge
symfony/twig-bundle                 v4.3.4     Symfony TwigBundle
symfony/validator                   v4.3.4     Symfony Validator Component
symfony/var-exporter                v4.3.4     A blend of var_export() + serialize() to turn any serializable data structure to plain PHP ...
symfony/web-link                    v4.3.4     Symfony WebLink Component
symfony/yaml                        v4.3.4     Symfony Yaml Component
theseer/tokenizer                   1.1.3      A small library for converting tokenized PHP source code into XML and potentially other for...
twig/twig                           v2.11.3    Twig, the flexible, fast, and secure template language for PHP
webmozart/assert                    1.5.0      Assertions to validate method input/output with nice error messages.
willdurand/negotiation              v2.3.1     Content Negotiation tools for PHP provided as a standalone library.
zendframework/zend-code             3.3.2      Extensions to the PHP Reflection API, static code scanning, and code generation
zendframework/zend-eventmanager     3.2.1      Trigger and listen to events within a PHP application
* Suite configuration:

```yml
# paste suite config here

actor: FunctionalTester
modules:
    enabled:
        - Symfony:
            app_path: 'src'
            environment: 'test'
        - Doctrine2:
            depends: Symfony
            cleanup: true
        - \App\Tests\Helper\Functional
        - REST:
            depends: Symfony

```

**Comment by Naktibalda** [comment]
@Basster @alexkunin Please take a look at this issue.
It looks like it was caused by https://github.com/Codeception/Codeception/pull/5685

**Comment by Basster** [comment]
Hi. I've created #5692 to be able to send non-entity doctrine ids, through the debug message. Nevertheless this also looks like some case of missing namespace import.

@erop maybe you have to add `Doctrine\Common\Annotations\AnnotationRegistry::registerLoader('class_exists');` to your `_bootstrap.php` file, to load the `Ramsey\Uuid\Uuid` class properly.

**Comment by erop** [comment]
Sorry, it doesn't help. You may check it using this repo https://github.com/erop/ramsey-uuid-in-codeception

**Comment by Basster** [comment]
Yes, no workaround possible in 3.1.0. I've fixed it in the PR.

**Comment by Naktibalda** [comment]
Fixed by #5692
